model/MMA8453.cpp

```diff
@@ -0,0 +1,255 @@
+#include "MMA8453.h"
+#include "CodalConfig.h"
+#include "ErrorNo.h"
+#include "CodalCompat.h"
+#include "CodalFiber.h"
+
+using namespace codal;
+
+//
+// Configuration table for available g force ranges.
+// Maps g -> LIS3DH_CTRL_REG4 [5..4]
+//
+static const KeyValueTableEntry accelerometerRangeData[] = {
+    {2, 0},
+    {4, 1},
+    {8, 2},
+ 
+};
+CREATE_KEY_VALUE_TABLE(accelerometerRange, accelerometerRangeData);
+
+
+/**
+  * Constructor.
+  * Create a software abstraction of an accelerometer.
+  *
+  * @param _i2c an instance of DeviceI2C used to communicate with the onboard accelerometer.
+  * @param _int1 the pin connected to the LIS3DH IRQ line.
+  * @param coordinateSpace The orientation of the sensor.
+  * @param address the default I2C address of the accelerometer. Defaults to: LIS3DH_DEFAULT_ADDR.
+  * @param id the unique EventModel id of this component. Defaults to: DEVICE_ID_ACCELEROMETER
+  *
+  * @code
+  * DeviceI2C i2c = DeviceI2C(I2C_SDA0, I2C_SCL0);
+  *
+  * LIS3DH accelerometer = LIS3DH(i2c);
+  * @endcode
+ */
+MMA8453::MMA8453(Pin& sda, Pin& scl, Pin &_int1, CoordinateSpace &coordinateSpace, uint16_t address,  uint16_t id) : Accelerometer(coordinateSpace, id), i2c(sda, scl), int1(_int1), sample()
+{
+    // Store our identifiers.
+    this->id = id;
+    this->status = 0;
+    this->address = address;
+
+    // Update our internal state for 50Hz at +/- 2g (50Hz has a period af 20ms).
+    this->samplePeriod = 20;
+    this->sampleRange = 2;
+
+    // Configure and enable the accelerometer.
+    configure();
+}
+
+/**
+  * Configures the accelerometer for G range and sample rate defined
+  * in this object. The nearest values are chosen to those defined
+  * that are supported by the hardware. The instance variables are then
+  * updated to reflect reality.
+  *
+  * @return DEVICE_OK on success, DEVICE_I2C_ERROR if the accelerometer could not be configured.
+  */
+int MMA8453::configure()
```

Comment:
Go through this whole file to make sure it has the correct and consistent brace and newline style